src/components/tabs/LayersTab.vue

```diff
@@ -25,19 +31,37 @@ import AddNodeButton from '@/components/buttons/AddNodeButton.vue';
 import ButtonGrid from '@/components/buttons/ButtonGrid.vue';
 import { mapGetters } from 'vuex';
 import EditorManager from '@/EditorManager';
-import { ModelCategories } from '@/nodes/model/Types';
+import { ModelCategories, ModelNodes } from '@/nodes/model/Types';
+import SearchBar from '@/SearchBar.vue';
 
 @Component({
   components: {
     ExpandablePanel,
     AddNodeButton,
     ButtonGrid,
+    SearchBar,
   },
   computed: mapGetters(['editorIONames']),
 })
 export default class LayersTab extends Vue {
-  private modelNodes = EditorManager.getInstance().modelCanvas.nodeList;
-  private ioNodes = this.modelNodes[0];
-  private ioLabel = ModelCategories.IO;
+  private nodeList = EditorManager.getInstance().modelCanvas.nodeList;
+  private modelCategories = ModelCategories;
+  private modelNodeTypes = ModelNodes;
+  private searchString = '';
+
+  private get renderedNodes() {
+    return this.nodeList.map((section) => ({
+      category: section.category,
+      nodes: section.nodes.filter((node) => this.shouldRender(node.name)),
+    }));
+  }
+
+  private search(search: string) {
+    this.searchString = search;
+  }
+
+  private shouldRender(button: string) {
+    return button.toLowerCase().includes(this.searchString.toLowerCase());
+  }
```

Comment:
Can we extract this to a common method that all the classes can use? We would have to pass in `searchString`

Reply:
Ik that's why I didn't, because then you have to pass it in from the `template` as well. Personally I don't like haveing to do that, do you think I should tho?

Reply:
Given the already existing duplication that ideally would be cleared up, it seems a bit weird that we'd clean only a small bit up, but idk

Reply:
Plus, it might be that in the `customTab` we would want to show all the functions if you search for a file instead, so you could change just that specific `shouldRender` function, maybe. Pretty niche use case for that tho, lol